ee/secureenclavesigner/secureenclavesigner_darwin.go

```diff
@@ -0,0 +1,295 @@
+//go:build darwin
+// +build darwin
+
+package secureenclavesigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"os/user"
+	"strings"
+	"sync"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/allowedcmd"
+	"github.com/kolide/launcher/ee/consoleuser"
+	"github.com/kolide/launcher/pkg/traces"
+)
+
+const (
+	CreateKeyCmd     = "create-key"
+	PublicEccDataKey = "publicEccData"
+)
+
+type opt func(*secureEnclaveSigner)
+
+type secureEnclaveSigner struct {
+	uidPubKeyMap         map[string]*ecdsa.PublicKey
+	pathToLauncherBinary string
+	store                types.GetterSetterDeleter
+	slogger              *slog.Logger
+	mux                  *sync.Mutex
+}
+
+func New(ctx context.Context, slogger *slog.Logger, store types.GetterSetterDeleter, opts ...opt) (*secureEnclaveSigner, error) {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	ses := &secureEnclaveSigner{
+		uidPubKeyMap: make(map[string]*ecdsa.PublicKey),
+		store:        store,
+		slogger:      slogger.With("component", "secureenclavesigner"),
+		mux:          &sync.Mutex{},
+	}
+
+	data, err := store.Get([]byte(PublicEccDataKey))
+	if err != nil {
+		traces.SetError(span, fmt.Errorf("getting public ecc data from store: %w", err))
+		return nil, fmt.Errorf("getting public ecc data from store: %w", err)
+	}
+
+	if data != nil {
+		if err := json.Unmarshal(data, ses); err != nil {
+			traces.SetError(span, fmt.Errorf("unmarshaling secure enclave signer: %w", err))
+			ses.slogger.Log(ctx, slog.LevelError,
+				"unable to unmarshal secure enclave signer, data may be corrupt, wiping",
+				"err", err,
+			)
+
+			if err := store.Delete([]byte(PublicEccDataKey)); err != nil {
+				traces.SetError(span, fmt.Errorf("deleting corrupt public ecc data: %w", err))
+				return nil, fmt.Errorf("deleting corrupt public ecc data: %w", err)
+			}
+		}
+	}
+
+	for _, opt := range opts {
+		opt(ses)
+	}
+
+	if ses.pathToLauncherBinary == "" {
```

Comment:
Feels a little odd to record this. Why not look at `os.Executable` each time?

Reply:
This is for testing. If we look at os.Executable in a test, it will have a go test binary and will error out.

Reply:
I will leave a comment about this

Reply:
comment left in https://github.com/kolide/launcher/pull/1647